### PR TITLE
Loot Tracker: Allow loot boxes to be collapsed

### LINKED ISSUE #9416: Add Collapsable Loot Tracker Entries
**Feature Description**
Add the ability to collapse per monster loot in the loot tracker plugin.

**Describe the solution you'd like**
Left-clicking the monsters name in the loot tracker should "collapse" the items looted for that monster. Left-click again to reveal again.

**Additional context**
If you're asking why this is useful, well I just finished a full day of slayer and my loot tracker scrolls for days. I want to shorten that list so it's easier to navigate (for example: i want to find all of my loot for the brimstone chest), while keeping my "total value" at the top the actual total value. I can't use "reset" for that.

**Example Screenshots:** https://imgur.com/a/CCwr7b3

### PR DESCRIPTION
**Pull request by Serpa88**
Closes #9416 

image
image

### CODE REVIEW

**Comment by deathbeam** [comment]
I would just remove the collapse/uncollapse button, because it looks horrible. Also, maybe the headers of collapsed boxes should be dimmed?

**Comment by LuisYenial** [comment]
HE

**Comment by xdesr** [comment]
Instead of the colossal button perhaps it would be best to add a small button next to the other three i.e.

image

**Comment by Serpa88** [comment]
> Instead of the colossal button perhaps it would be best to add a small button next to the other three i.e.
> 
> image

Yeah this is exactly what I’m going to do. I do like that icon a lot, can you supply me the file?

**Comment by Serpa88** [comment]
I fixed the big button and added the dimming on the headers that @deathbeam suggested

**Comment by deathbeam** [comment]
Looks great

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerBox.java [hunk 1]
```diff
@@ -182,6 +184,48 @@ void rebuild()
...
+		if (isCollapsed())
+		{
+			itemContainer.setVisible(true);
+			applyDimmer(true, logTitle);
+		}
+	}
+
+	boolean isCollapsed()
+	{
+		return itemContainer.isVisible() == false;
```
Comment (psikoi) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerBox.java]: return  !itemContainer.isVisible()

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerBox.java [hunk 2]
```diff
@@ -182,6 +184,48 @@ void rebuild()
...
+	}
+
+	private void applyDimmer(boolean brighten, JPanel panel)
+	{
+		for (Component component : panel.getComponents())
+		{
+			Color color = component.getForeground();
+
+			if (brighten)
```
Comment (psikoi) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerBox.java]: I'd simplify this with a ternary

..setForeground(brighten ? color.brighter() : color.darker());

**Comment by psikoi** [comment]
Any chance you could replace the arrows with these?

 https://material.io/resources/icons/?search=arrow&icon=keyboard_arrow_down&style=baseline

I'd prefer a slimmer icon, would stand out less, i think

**Comment by Serpa88** [comment]
Made the changes requested

**Comment by psikoi** [comment]
Can't test but visually and code-wise it looks good

**Pull request merged**